spec/mongo/socket/ssl_spec.rb

```diff
@@ -279,6 +279,42 @@
       end
     end
 
+    context 'when the hostname is incorrect' do
+      let(:host_name) do
+        'incorrect_hostname'
+      end
+
+      context 'when the hostname is verified' do
+        it 'raises an error' do
+          expect {
+            described_class.new(
+              address.host,
+              address.port,
+              host_name,
+              30,
+              ::Socket::PF_INET,
+              options.merge(tls_allow_invalid_hostnames: false)
+            ).connect!
+          }.to raise_error(Mongo::Error::SocketError)
```

Comment:
This assertion should include something about the hostname not passing verification. As written it will be satisfied by failing to DNS resolve `incorrect_hostname`.

Reply:
I don't believe this is true; if you look at the implementation of `Mongo::Socket::SSL`, you'll see that the socket is connected with the first two arguments (which are `address.host` and `address.port`), not with the hostname. Furthermore, when changing line 298 to assert `not_to raise_error`, the error raised is specifically the one the driver raises in `Mongo::Socket::SSL#verify_certificate!` (not related to DNS), meaning that the socket connected to the server properly.

Reply:
Can we add the message assertion here?